modules/frontend/tracebyidsharding.go

```diff
@@ -175,8 +176,10 @@ func (s shardQuery) RoundTrip(r *http.Request) (*http.Response, error) {
 	}
 
 	return &http.Response{
-		StatusCode:    http.StatusOK,
-		Header:        http.Header{},
+		StatusCode: http.StatusOK,
+		Header: http.Header{
+			api.HeaderContentType: {api.HeaderAcceptProtobuf},
```

Comment:
it's weird that we assume protobuf here b/c of a header that was set in frontend.go.

Reply:
It's not, the Trace Sharder [_only_ ever sends a protobuf message](https://github.com/grafana/tempo/blob/main/modules/frontend/tracebyidsharding.go#L166-L171), it's then unmarshalled and re-marshalled back into either protobuf or json in the `frontend.go` in the `newTraceByIDMiddleware` function [here](https://github.com/grafana/tempo/blob/main/modules/frontend/frontend.go#L137-L151)

Reply:
My bad, I meant it's weird we assume that the data received in the trace by id sharder (from the querier) [is proto](https://github.com/grafana/tempo/blob/7d26ee26986411cf096f4a93ad1dd7fa422fa972/modules/frontend/tracebyidsharding.go#L120-L121), but the only reason it is proto is because a [header that's set](https://github.com/grafana/tempo/blob/7d26ee26986411cf096f4a93ad1dd7fa422fa972/modules/frontend/frontend.go#L116) in frontend.go. 

I think we should set the `Accept` for proto in `tracebyidsharding.go` since that code also assumes that the returned data is proto. But these gross corners have been around for long before you made this improvement.

Reply:
Oh I see! I assumed this was intentional after reading the comment on that same line you linked, I guess it kinda make sense to keep content negotiation as close the edge as possible? Then again if internal communication is expected to be only protobuf, then setting the Accept header might just be superfluous 🤷